src/shared/use-data-value-set/use-data-value-set.js

```diff
@@ -64,6 +64,13 @@ export const useDataValueSet = () => {
         // Only enable this query if there are no ongoing mutations
         enabled: isValidSelection,
         select,
+        refetchOnMount: (query) => {
+            // only refetch on mount if the query was just hydrated
+            // this should only happen during initial app-load.
+            // If we were to return 'false' the query would not be refetch on first mount
+            // because it's mounted with hydrated-state
+            return query.meta.isHydrated
+        },
```

Comment:
Related to my previous comment, I presently failing to see how `isHydrated` would ever get any other value than `true` here.

Reply:
When the `dataValueSet` is updated, it will not have that property because the entire query is "new" and overridden.